modules/web/src/de/diedavids/cuba/wizard/web/screens/examples/example1/WizardExample1.java

```diff
@@ -28,5 +35,14 @@ public void ready() {
                     event.getPrevStep().getId() + " to " +
                     event.getStep().getId());
         });
+
+
+        WizardStepAware wizardStepAware = (WizardStepAware) openFrame(null, "example-1-step-4-frame");
+
+
+        WizardStep wizardStep = wizard.addStep(3, "Hello", wizardStepAware);
+        wizardStep.setId("Hello");
+        wizardStep.setCaption("Hello 123");
```

Comment:
if I got it right, the problem is that when I want to do this, the wizardStep instance needs access to the `TabSheet.Tab` instance. But currently I don't see a way on how to do it - because currently the reference chain is the other way around: 
https://github.com/mariodavid/cuba-component-wizard/pull/3/commits/bb9bfaa440ef0172e4ec26eaf54681422163ccda#diff-87ace81352c45af9cbca75363506d205R371

Reply:
You are setting properties of WebVBoxLayout. Those properties are not automatically reflected to the corresponding Tab. I suppose that you should store reference on Tab in WizardStep and update caption and icon of the Tab.

Reply:
There is an more circumstance: SplitPanel and TabSheet do not paint caption and icon of child components. That's why you do not see your caption/icon anywhere. As you probably know, in Vaadin, Layout components manage captions, icons and positions of child components.

Reply:
hi, thanks  @jreznot for looking at it.

I think I did not clearly mentioned what my problem is. You are right, as it is right now it does not work. My question was more how can I make it happen.

Wizard API wise I would like it to be like this:
```
WizardStep wizardStep = wizard.addStep(3, "Hello", wizardStepAware);
wizardStep.setId("Hello");
wizardStep.setCaption("Hello 123");
```
Just as it is with the `TabSheet.Tab` directly. 

Here's the problem that I'm facing in the implementation of this API. My idea was to override `setId`, `setCaption` & `serIcon` etc. and delegate those calls to the `TabSheet.Tab` reference of that wizard step. I assume that this will do the trick. 
Actually I already implemented a [TabSheet.Tab field](https://github.com/mariodavid/cuba-component-wizard/commit/bb9bfaa440ef0172e4ec26eaf54681422163ccda#diff-e0c4ff239857dc7827ba4fe4ec3c1024R12) in the WebWizardStep, but did not really use it, becuase I faced the following problem:

[Here](https://github.com/mariodavid/cuba-component-wizard/blob/bb9bfaa440ef0172e4ec26eaf54681422163ccda/modules/web/src/de/diedavids/cuba/wizard/web/gui/components/WebWizard.java#L371) where the `TabSheet.Tab` is created, the `wizardStep` instance already exists, because it need to be passed into the content as the child component in `addTab`.

In order to solve that bi-directional reference problem I thought of two options:

1. I'll add a method to the `WizardStep` interface: `setWrapperComponent` or something and call it [here](https://github.com/mariodavid/cuba-component-wizard/commit/bb9bfaa440ef0172e4ec26eaf54681422163ccda#diff-87ace81352c45af9cbca75363506d205R372). This way the reference can get passed to the WIzardStep instance. Then the delegation will work (as I assumed above)

2. I'll re-think if the current way how the `WebWizardStep` is layed out is even correct. E.g. the fact that it extends from `WebVBoxLayout` happened more by accident and not by intention. So I might have taken a wrong path and therefore are facing this problem.

I think 1. will work but I'm not sure if this is a proper solution or what other downsides (especially when it is used in the case of the XML loader) may come up due to this. Regarding 2. I don't really have much experience in order to see the full picture here so that I would be qualified to see alternatives. 

Perhaps you might elaborate on those options. Thanks.

Reply:
I cannot say which way is better, it is all about your design decisions. I do see any blockers here and I'd simply go with the first option.

By the way, I've noticed that you use a lot of empty lines in code. This makes code less readable and it looks like sieve.

Reply:
ok thanks @jreznot - did it as you mentioned.